pkg/controllers/management/auth/globalroles/globalrolebinding_handler.go

```diff
@@ -411,9 +572,11 @@ func (grb *globalRoleBindingLifecycle) grantRestrictedAdminUserClusterPermission
 	return returnErr
 }
 
-func IsClusterUnavailable(err error) bool {
-	if apiError, ok := err.(*httperror.APIError); ok {
-		return apiError.Code == httperror.ClusterUnavailable
-	}
-	return false
+// isCRTBValid determines if a given CRTB is up to date for a given cluster and owning global role binding. Should
+// only be used in the context of CRTBs owned by GRBs
+func isCRTBValid(crtb *v3.ClusterRoleTemplateBinding, cluster *v3.Cluster, binding *v3.GlobalRoleBinding) bool {
```

Comment:
I see there are no nil checks for cluster, crtb, grb, as they flow from function to function, which is fine, since they are returned from caches, and there is error handling. But maybe adding a `nil || nil || nil => false` might be in order, not sure...

Reply:
If these were exported/used outside of this package I would agree with you on that. But since these are only used internally, I don't think that this is necessary.

Reply:
Since this is not public, it is okay to leave it as is, but it is good practice to always handle the default value of objects. Similar to how you can call `len` and `range` on a `nil` slice. 
I will note that I do not always do this on my PRs, so I will not press the issue.

Reply:
I've adjusted to meet what both of you suggested, please give it a look and lmk if you see anything else that would be helpful.